### ISSUE TITLE
Structure view

### ISSUE THREAD

**Issue by bjansen** [ceylon/ceylon-ide-intellij#70]
(a.k.a Outline in eclipse)

Already done:

* imports
* classes, interfaces
* methods

To do:

* top level values, attributes
* objects
* inherited members
* hide unshared
* icons for unshared, variable etc.

**Comment by bjansen** [comment]
A few problems:

* hide unshared members doesn't work in the structure popup (it hides things if not checked), but that's a problem that I fixed in IntelliJ, so we'll wait for the fix to be released (https://github.com/JetBrains/intellij-community/pull/285)
* syntax highlighting is not yet possible, I submitted a PR to add the feature (https://github.com/JetBrains/intellij-community/pull/286)
* imports should be collapsed by default, but that works only in the tool window, not the popup
* inherited members are shown only once in the popup, so if a file contains two classes, members inherited from `Object` will be shown in the first class only

**Comment by bjansen** [comment]
My work here is done, the only things left depend on fixes from JetBrains.

**Comment by gavinking** [comment]
Cool!

On Thu, Jul 9, 2015 at 4:20 PM, Bastien Jansen <notifications@github.com>
wrote:

> My work here is done, the only things left depend on fixes from JetBrains.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ceylon/ceylon-ide-intellij/issues/70#issuecomment-119998886>
> .
>



-- 
Gavin King
gavin@ceylon-lang.org
http://profiles.google.com/gavin.king
http://ceylon-lang.org
http://hibernate.org
http://seamframework.org

**Comment by bjansen** [comment]
Since the only things left to do rely on fixes in IntelliJ, I'm closing this one. See https://github.com/ceylon/ceylon-ide-intellij/issues/76 for the remaining problems.
